code/network/multi_obj.cpp

```diff
@@ -1890,89 +1894,123 @@ int multi_oo_unpack_data(net_player* pl, ubyte* data, int seq_num, int time_delt
-					vm_extract_angles_matrix_alternate(angs_2, &subsysp->submodel_instance_2->canonical_orient);
-				}
+				// We could do individual flag comparisons, but every other flag is animation, and health alone is the lowest value.
+				if (flags[i] > OO_SUBSYS_HEALTH){
 
-				if (flags[i] & OO_SUBSYS_ROTATION_1b) {
-					prev_angs_1->b = angs_1->b;
-					angs_1->b = (subsys_data[data_idx] * PI2);
-					data_idx++;
-				}
+					bool animations_valid = false;
 
-				if (flags[i] & OO_SUBSYS_ROTATION_1h) {
-					prev_angs_1->h = angs_1->h;
-					angs_1->h = (subsys_data[data_idx] * PI2);
-					data_idx++;
-				}
+					if (seq_num > pobjp->interp_info.get_subsystem_animation_frame(i)) {
+						animations_valid = true;
+						pobjp->interp_info.set_subsystem_animation_frame(i, seq_num);						
+					}
 
-				if (flags[i] & OO_SUBSYS_ROTATION_1p) {
-					prev_angs_1->p = angs_1->p;
-					angs_1->p = (subsys_data[data_idx] * PI2);
-					data_idx++;
-				}
+					angles prev_angs_1 = vmd_zero_angles;
+					angles prev_angs_2 = vmd_zero_angles;
+					angles angs_1 = vmd_zero_angles;
+					angles angs_2 = vmd_zero_angles;
 
-				if (flags[i] & OO_SUBSYS_ROTATION_2b) {
-					prev_angs_2->b = angs_2->b;
-					angs_2->b = (subsys_data[data_idx] * PI2);
-					data_idx++;
-				}
+					if (subsysp->submodel_instance_1) {
+						vm_extract_angles_matrix_alternate(&prev_angs_1, &subsysp->submodel_instance_1->canonical_prev_orient);
+						vm_extract_angles_matrix_alternate(&angs_1, &subsysp->submodel_instance_1->canonical_orient);
+					}
 
-				if (flags[i] & OO_SUBSYS_ROTATION_2h) {
-					prev_angs_2->h = angs_2->h;
-					angs_2->h = (subsys_data[data_idx] * PI2);
-					data_idx++;
-				}
+					if (subsysp->submodel_instance_2) {
+						vm_extract_angles_matrix_alternate(&prev_angs_2, &subsysp->submodel_instance_2->canonical_prev_orient);
+						vm_extract_angles_matrix_alternate(&angs_2, &subsysp->submodel_instance_2->canonical_orient);
+					}
 
-				if (flags[i] & OO_SUBSYS_ROTATION_2p) {
-					prev_angs_2->p = angs_2->p;
-					angs_2->p = (subsys_data[data_idx] * PI2);
-					data_idx++;
-				}
+					if (flags[i] & OO_SUBSYS_ROTATION_1b) {
+						if (animations_valid) {
+							prev_angs_1.b = angs_1.b;
+							angs_1.b = (subsys_data[data_idx] * PI2);
+						}
 
-				if (flags[i] & OO_SUBSYS_TRANSLATION_x) {
-					subsysp->submodel_instance_1->canonical_prev_offset.xyz.x = subsysp->submodel_instance_1->canonical_offset.xyz.x;
-					subsysp->submodel_instance_1->canonical_offset.xyz.x = subsys_data[data_idx];
-					data_idx++;
-				}
+						data_idx++;
+					}
 
-				if (flags[i] & OO_SUBSYS_TRANSLATION_y) {
-					subsysp->submodel_instance_1->canonical_prev_offset.xyz.y = subsysp->submodel_instance_1->canonical_offset.xyz.y;
-					subsysp->submodel_instance_1->canonical_offset.xyz.y = subsys_data[data_idx];
-					data_idx++;
-				}
+					if (flags[i] & OO_SUBSYS_ROTATION_1h) {
+						if (animations_valid) {						
```

Comment:
Super minor, but might not need all that extra whitespace to the right.

Reply:
![image](https://user-images.githubusercontent.com/14077810/192415840-0178e5a5-c17e-466c-a6fe-1f0b89bed749.png)